lib/collection/src/shards/local_shard/clock_map.rs

```diff
@@ -254,6 +254,48 @@ impl From<api::grpc::qdrant::RecoveryPoint> for RecoveryPoint {
     }
 }
 
+#[derive(Clone, Debug, Deserialize, Serialize)]
+struct ClockMapHelper {
+    clocks: Vec<KeyClockHelper>,
+}
+
+impl From<ClockMap> for ClockMapHelper {
+    fn from(clock_map: ClockMap) -> Self {
+        Self {
+            clocks: clock_map.clocks.into_iter().map(Into::into).collect(),
+        }
+    }
+}
+
+impl From<ClockMapHelper> for ClockMap {
+    fn from(helper: ClockMapHelper) -> Self {
+        Self {
+            clocks: helper.clocks.into_iter().map(Into::into).collect(),
+        }
+    }
+}
+
+#[derive(Copy, Clone, Debug, Deserialize, Serialize)]
+struct KeyClockHelper {
+    #[serde(flatten)]
+    key: Key,
+
+    #[serde(flatten)]
```

Comment:
I think this fails because `Clock` has transparent.

I guess the best would be to remove transparent, and keep flatten here.

Reply:
Yeah, you're probably right. Nice catch.